### ISSUE TITLE
[1.2.0pre2] Context save doesn't populate extension inside TweetDeck in Firefox

### ISSUE THREAD

**Issue by Simon-Tesla** [Simon-Tesla/RaccoonyWebEx#20]
Files are saved without any extension from inside TweetDeck (which has no plugin). Need to investigate whether the fetch to detect mime-type is erroring out.

Console log
```
[raccoony] initializing plugin tweetdeck.twitter.com contextMenuInject.js:216:38
[raccoony] got media 
{…}
​
downloadDestination: 1
​
filename: "Dg4awIDUYAAovi-"
​
siteFilename: "Dg4awIDUYAAovi-.jpg:large"
​
siteName: "tweetdeck.twitter.com"
​
sourceUrl: "https://tweetdeck.twitter.com/"
​
type: "image"
​
url: "https://pbs.twimg.com/media/Dg4awIDUYAAovi-.jpg:large"
​
<prototype>: Object { … }
contextMenuInject.js:216:38
[raccoony] sending message 
{…}
​
action: "messageaction_download"
​
data: {…}
​​
downloadDestination: 1
​​
filename: "Dg4awIDUYAAovi-"
​​
siteFilename: "Dg4awIDUYAAovi-.jpg:large"
​​
siteName: "tweetdeck.twitter.com"
​​
sourceUrl: "https://tweetdeck.twitter.com/"
​​
type: "image"
​​
url: "https://pbs.twimg.com/media/Dg4awIDUYAAovi-.jpg:large"
​​
<prototype>: Object { … }
​
<prototype>: Object { … }
contextMenuInject.js:216:38
Source map error: TypeError: NetworkError when attempting to fetch resource.
Resource URL: moz-extension://25e31232-b861-4507-a5fa-aa3bdfde6bc9/contextMenuInject.js
Source Map URL: filenames.js.map[Learn More]
```

**Comment by momoe** [comment]
I can verify this issue.

**Comment by Simon-Tesla** [comment]
I'll take a look, though Hentai Foundry has a normal page plugin now so I'd recommend using that over the context menu for downloading stuff. 

I did a little bit of work to try to make the context menu recognize if it could use the existing page plugin code to handle the download and switch over to that code instead of doing its default context handling, but it's not 100% reliable.

**Comment by Simon-Tesla** [comment]
Looks like a CORS error, should be solved by giving the `<all_urls>` permission to the extension.

**Comment by Simon-Tesla** [comment]
Fixed in 1.2.1
